File: src/main/java/back/ecommerce/cart/dto/response/CartProducts.java
package back.ecommerce.cart.dto.response;

import java.util.Collections;
import java.util.List;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@AllArgsConstructor(access = AccessLevel.PROTECTED)
@NoArgsConstructor
@Getter
public class CartProducts {
	private int count;
	private Long totalPrice;
	private List<CartProductDto> value;

	public static CartProducts create(List<CartProductDto> list) {
		long totalPrice = Collections.unmodifiableList(list).stream()
			.mapToLong(CartProductDto::getPrice)
			.sum();
		return new CartProducts(list.size(), totalPrice, list);
	}
}

File: src/main/java/back/ecommerce/cart/dto/response/CartProducts.kt
package back.ecommerce.cart.dto.response

import java.util.*

data class CartProducts(
    val count: Int,
    val totalPrice: Long,
    val value: List<CartProductDto>
) {
    companion object {
        fun create(list: List<CartProductDto>): CartProducts {
            val unmodifiableList = Collections.unmodifiableList(list)
            val totalPrice = unmodifiableList.stream()
                .mapToLong { it.price * it.quantity }
                .sum()
            val totalCount = unmodifiableList.stream()
                .mapToInt {
                    it.quantity
                }.sum()
            return CartProducts(totalCount, totalPrice, list)
        }
    }
}
